lua/null-ls/builtins/diagnostics/golangci_lint.lua

```diff
@@ -20,13 +20,40 @@ return h.make_builtin({
         ignore_stderr = true,
         multiple_files = true,
         cwd = h.cache.by_bufnr(function(params)
-            return u.root_pattern("go.mod")(params.bufname)
+            -- find the nearest config and use that directory since v2 defaults to
+            -- "relative-path-mode: cfg" which reports file names relative to
+            -- the directory of the config file
+            local cfg_path_yml = u.root_pattern(".golangci.yml")(params.bufname)
+            if cfg_path_yml then
+                return cfg_path_yml
+            end
+            local cfg_path_yaml = u.root_pattern(".golangci.yaml")(params.bufname)
+            if cfg_path_yaml then
+                return cfg_path_yaml
+            end
+            local cfg_path_toml = u.root_pattern(".golangci.toml")(params.bufname)
+            if cfg_path_toml then
+                return cfg_path_toml
+            end
+            local cfg_path_json = u.root_pattern(".golangci.json")(params.bufname)
+            if cfg_path_json then
+                return cfg_path_json
+            end
+            -- nil defaults to git root
+            return nil
+        end),
+        args = h.cache.by_bufnr(function(params)
+            -- params.command respects prefer_local and only_local options
```

Comment:
While this appears to be true from the source code, it is not working in my testing yet. I'm am currently debugging.